lua/rustaceanvim/config/server.lua

```diff
@@ -22,13 +22,15 @@ function server.load_rust_analyzer_settings(project_root, opts)
   local default_settings = opts.default_settings or config.server.default_settings
   local use_clippy = config.tools.enable_clippy and vim.fn.executable('cargo-clippy') == 1
   ---@diagnostic disable-next-line: undefined-field
-  if default_settings['rust-analyzer'].checkOnSave == nil and use_clippy then
+  if default_settings['rust-analyzer'].check == nil and use_clippy then
```

Comment:
issue: We should also take into account that somebody could be using `checkOnSave` 

```suggestion
  if default_settings['rust-analyzer'].check == nil and use_clippy and type(default_settings['rust-analyzer'].checkOnSave) ~= 'table' then
```